dali/operators/util/reshape.cc

```diff
@@ -132,33 +189,64 @@ void Reshape<Backend>::CalculateOutputShape(const Workspace &ws) {
   const int N = input_shape_.num_samples();
   switch (shape_source_) {
     case ShapeSource::Arg:
-      if (output_shape_.num_samples() != N) {
-        output_shape_ = uniform_list_shape(N, uniform_shape_);
+      if (use_rel_shape_) {
+        output_shape_.resize(N, rel_uniform_shape_.size());
+        for (int i = 0; i < N; i++) {
+          for (int d = 0; d < output_shape_.sample_dim(); d++) {
+            int out_e = round_int(rel_uniform_shape_[d] * input_shape_.tensor_shape_span(i)[d]);
+            output_shape_.tensor_shape_span(i)[d] = out_e;
+          }
+        }
+      } else {
+        if (output_shape_.num_samples() != N) {
+          output_shape_ = uniform_list_shape(N, uniform_shape_);
+        }
       }
       break;
     case ShapeSource::ArgInput:
-      ShapeFromInput(ws.ArgumentInput("shape"));
+      if (use_rel_shape_)
+        ShapeFromInput(ws.ArgumentInput("shape"), false);
+      else
+        ShapeFromInput(ws.ArgumentInput("rel_shape"), false);
       break;
     case ShapeSource::Input:
-      ShapeFromInput(ws.template InputRef<CPUBackend>(1));
+      ShapeFromInput(ws.template InputRef<CPUBackend>(1), use_rel_shape_);
```

Comment:
This won't ever work with relative shapes provided as inputs, as even if we get second input with float values, we won't set `use_rel_shape_=true`.

Reply:
Also, does it actually work on GPU to receive the shapes as CPU input? (It's not like it will be widely used due to it's limitations).

Reply:
1. It will always be false.
2. It does work and is actually tested.